src/rgb/psbt.rs

```diff
@@ -473,66 +475,84 @@ where
 pub trait PsbtInputEx<T> {
     type Error: std::error::Error;
 
-    fn resolve_psbt_input(
+    fn from_request(
         psbt_input: PsbtInputRequest,
         descriptor: Descriptor<DerivationAccount>,
-        wallet: Option<RgbWallet>,
         tx_resolver: &impl ResolveTx,
+        wallet: Option<RgbWallet>,
+        options: Option<NewPsbtOptions>,
     ) -> Result<T, Self::Error>;
 }
 
 impl PsbtInputEx<InputDescriptor> for InputDescriptor {
     type Error = PsbtInputError;
 
-    fn resolve_psbt_input(
+    fn from_request(
         psbt_input: PsbtInputRequest,
         descriptor: Descriptor<DerivationAccount>,
-        wallet: Option<RgbWallet>,
         tx_resolver: &impl ResolveTx,
+        wallet: Option<RgbWallet>,
+        options: Option<NewPsbtOptions>,
     ) -> Result<Self, Self::Error> {
-        let sighash: PsbtSigHashRequest = psbt_input.sigh_hash.unwrap_or_default();
+        let opt = options.unwrap_or_default();
         let outpoint: OutPoint = psbt_input.utxo.parse().expect("invalid outpoint parse");
 
-        let mut input: InputDescriptor = InputDescriptor {
+        let sighash: PsbtSigHashRequest = psbt_input.sigh_hash.unwrap_or_default();
+        let sighash_type = EcdsaSighashType::from_consensus(sighash as u32);
+
+        let terminal = psbt_input
+            .utxo_terminal
+            .parse::<DerivationSubpath<UnhardenedIndex>>()
+            .map_err(|_| PsbtInputError::WrongTerminal)?;
+
+        let seq_no = if opt.rbf {
+            SeqNo::rbf()
+        } else {
+            SeqNo::default()
+        };
+
+        let mut input = InputDescriptor {
             outpoint,
-            terminal: psbt_input
-                .utxo_terminal
-                .parse::<DerivationSubpath<UnhardenedIndex>>()
-                .map_err(|_| PsbtInputError::WrongTerminal)?,
-            seq_no: SeqNo::default(),
+            terminal,
+            seq_no,
+            sighash_type,
             tweak: None,
-            sighash_type: EcdsaSighashType::from_consensus(sighash as u32),
         };
 
         // Verify TapTweak (User Input or Watcher inspect)
-        if let Some(tapret) = psbt_input.tapret {
-            input.tweak = Some((
-                Fingerprint::default(),
-                tapret
-                    .parse::<sha256::Hash>()
-                    .map_err(|op| PsbtInputError::WrongInputTweak(op.to_string()))?,
-            ))
-        } else if let Some(tweak) = complete_input_desc(
-            descriptor.clone(),
-            input.clone(),
-            wallet.clone(),
-            tx_resolver,
-        )
-        .map_err(|op| PsbtInputError::WrongWatcherTweak(op.to_string()))?
-        {
-            input.tweak = Some((
-                Fingerprint::default(),
-                tweak
+        input.tweak = match psbt_input.tapret {
```

Comment:
Why are we doing our own verification here? Shouldn't we be delegating that to RGB wallet / core?